src/main/kotlin/com/mdrobnak/lalrpop/psi/LpResolveType.kt

```diff
@@ -16,5 +21,5 @@ interface LpResolveType : PsiElement {
      * And referenced with Nonterminal<A, B> in another symbol, the list of arguments should be the resolved types of
      * "A" and "B", in this order.
      */
-    fun resolveType(arguments: List<NonterminalGenericArgument>): String
+    fun resolveType(context: LpTypeResolutionContext, arguments: List<NonterminalGenericArgument>): String
```

Comment:
What do you think of having an overloaded `resolveType(arguments)` which calls `containingFile.lalrpopTypeResolutionContext()`?

Reply:
The whole point was to try not to find those types every time

Reply:
I think it's ok in this instance. I re-enabled the code locally and didn't see any issues. This isn't library code and it doesn't interface with Java, so I don't think there should be any issues.

Reply:
Well the generated classes for psi are all java :/

Reply:
Yeah, you're right about the generated classes. The extension function method works well enough, and it turns out that we don't really need the implicit context form that often anyways. You can either remove the implicit context form or use the extension method.